frontend/src/components/SearchForm.js

```diff
@@ -14,21 +14,42 @@ const SearchForm = () => (
 
             <div class="radio-bar-container">
                 <div class="radio-choice-container">
-                    <input type="radio" id="beginner" name="select" value="beginner" />
+                    <input 
+                        type="radio" 
+                        id="beginner" 
+                        name="select" 
+                        value="beginner" 
+                        checked={difficulty === 'beginner'} 
+                        onClick={() => setDifficulty('beginner')}
```

Comment:
Please, consider implementing the subscriber function pattern here. It will be easier to find the function setting the state that way.

Reply:
1 .So I should create function onDifficulty:

```
const onDifficulty = (value) => {
    onSetDifficulty(value);
}
```

2. And then create another function onSetDifficulty();
```
const onSetDifficulty = (value) => {
    setDifficulty(value);
}
```

Should they both be in main.js or the `onDifficulty` should be in the Searchform.js component?

Reply:
I'm sorry for not being very clear. In `Main` component, that is where you declare the state. So, you should create a function there that calls the `setRadioDifficulty` function to mutate the state. So, I have added some screenshots. Hope it helps.
![Screenshot from 2022-04-11 16-35-42](https://user-images.githubusercontent.com/64225540/162783733-c420a3bc-77d9-4995-9fb0-8e76425eeba7.png)
![Screenshot from 2022-04-11 16-35-25](https://user-images.githubusercontent.com/64225540/162783767-d9534511-7995-4753-9251-6919f7bd98a9.png)

Reply:
Thanks! That's very clear!